packages/alfa-rules/src/sia-r66/rule.ts

```diff
@@ -35,6 +37,34 @@ export default Rule.Atomic.of<Page, Text, Question.Metadata>({
   evaluate({ device, document }) {
     return {
       applicability() {
+        // Gather all aria-disabled widgets or groups on the document
+        const disabledWidgetTexts: Set<Text> = Set.from(
+          document
+            .descendants(Node.fullTree)
+            .filter((node) =>
+              test(
+                and(
+                  isElement,
+                  and(
+                    or(
+                      hasRole(device, (role) => role.isWidget()),
+                      hasRole(device, "group")
+                    ),
+                    isSemanticallyDisabled
+                  )
+                ),
+                node
+              )
```

Comment:
Looks like this can be eta-contracted and the call to `test` removed.
```suggestion
            .filter(and(
                  isElement,
                  and(
                    or(
                      hasRole(device, (role) => role.isWidget()),
                      hasRole(device, "group")
                    ),
                    isSemanticallyDisabled
                  )
                )
```